ios/Classes/DBManager.m

```diff
@@ -27,30 +27,38 @@ @implementation DBManager
 -(instancetype)initWithDatabaseFilePath:(NSString *)dbFilePath{
     self = [super init];
     if (self) {
-        // Set the documents directory path to the documentsDirectory property.
-        NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
-        self.documentsDirectory = [paths objectAtIndex:0];
+        // Set the app directory path to the application support files property.
+        self.appDirectory = [NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES) firstObject];
 
         // Keep the database filepath
         self.databaseFilePath = dbFilePath;
 
         // Keep the database filename.
         self.databaseFilename = [dbFilePath lastPathComponent];
 
-        // Copy the database file into the documents directory if necessary.
-        [self copyDatabaseIntoDocumentsDirectory];
+        // Copy the database file into the app directory if necessary.
+        [self copyDatabaseIntoAppDirectory];
     }
     return self;
 }
 
--(void)copyDatabaseIntoDocumentsDirectory{
-    // Check if the database file exists in the documents directory.
-    NSString *destinationPath = [self.documentsDirectory stringByAppendingPathComponent:self.databaseFilename];
+// Will be removed in the next major version.
+-(void)copyDatabaseIntoAppDirectory{
+    // Check if the database file exists in the app directory.
+    NSString *destinationPath = [self.appDirectory stringByAppendingPathComponent:self.databaseFilename];
     if (![[NSFileManager defaultManager] fileExistsAtPath:destinationPath]) {
-        // The database file does not exist in the documents directory, so copy it from the main bundle now.
-        NSString *sourcePath = self.databaseFilePath;
+
+        // Attemp database file migration from the documents directory if exists
+        NSString *documentsDirectory = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject];
+        NSString *migrationSourcePath = [documentsDirectory stringByAppendingPathComponent:self.databaseFilename];
         NSError *error;
-        [[NSFileManager defaultManager] copyItemAtPath:sourcePath toPath:destinationPath error:&error];
+        if ([[NSFileManager defaultManager] fileExistsAtPath:migrationSourcePath]) {
+            // Migrate the database file from the documents directory to the app directory
+            [[NSFileManager defaultManager] moveItemAtPath:migrationSourcePath toPath:destinationPath error:&error];    
+        } else {
+            // The database file does not exist in the app directory, so copy it from the main bundle now.
+            [[NSFileManager defaultManager] copyItemAtPath:self.databaseFilePath toPath:destinationPath error:&error];
```

Comment:
I strongly suspect that this is the wrong file path. On a fresh install of the example the file is not found. The `databaseFilePath` is not the root bundle path, is it?

Reply:
That's possible. I only accepted that person's PR after they confirmed it worked. If you have some time to fix this, then I'll be very grateful :)

Reply:
I did not touch the part of the code providing `databaseFilePath` value.
Previously, the value of this variable was stored and used in a local variable named `sourcePath` (line 51)

Reply:
@ndelanou yes, I saw that too. The file does not exist though. AppCode is warning me that the resource does not exist as well.

So this most likely has nothing to do with this PR. Maybe something new with iOS 16?

Reply:
Well, I cannot get this to work properly. 

Copying the file from the bundle is successful the first time around, but the `fileExistsAtPath` check always returns false afterwards. And actually opening the DB also always fails.

And I think there's a bug when checking if the DB could be opened.

Reply:
I know this is not the right place to have this discussion, but imho this whole DB thing has not worked since 2018 :/